server/logging/filesystem.go

```diff
@@ -24,8 +24,17 @@ type filesystemLogWriter struct {
 // NewFilesystemLogWriter creates a log file for osquery status/result logs.
 // The logFile can be rotated by sending a `SIGHUP` signal to Fleet if
 // enableRotation is true
+//
+// The enableCompression argument is only used when enableRotation is true.
 func NewFilesystemLogWriter(path string, appLogger log.Logger, enableRotation bool, enableCompression bool) (*filesystemLogWriter, error) {
 	if enableRotation {
+		// Fail early if the process does not have the necessary
+		// permissions to open the file at path.
+		file, err := openFile(path)
+		if err != nil {
+			return nil, errors.Wrap(err, "perm check")
+		}
+		file.Close()
```

Comment:
This is just a nitpick, because it doesn't make a lot of impact: We can probably pass this `file` to `newRawLogWriter` below.

Reply:
Good idea. I've pushed the change (and while I was at it I rearranged _NewFilesystemLogger_ a bit to decrease indentation).